lib/routes/baobua/article.ts

```diff
@@ -0,0 +1,54 @@
+import { load } from 'cheerio';
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+
+async function loadArticle(link) {
+    const resp = await got(link);
+    const article = load(resp.body);
+
+    const title = article('title')
+        .text()
+        .replace('BaoBua.Com:', '')
+        .replace(/\| Page \d+\/\d+/, '')
+        .trim();
+    const totalPagesRegex = /Page \d+\/(\d+)/;
+    const totalPagesMatch = totalPagesRegex.exec(article('title').text());
+    const totalPages = totalPagesMatch ? Number.parseInt(totalPagesMatch[1]) : 1;
+
+    let pubDate = new Date();
+    const jsonldScripts = article('script[type="application/ld+json"]').toArray();
+    for (const script of jsonldScripts) {
+        const jsonData = JSON.parse(load(script).text() || '');
+        if (jsonData['@type'] === 'BlogPosting') {
+            pubDate = jsonData.datePublished ? parseDate(jsonData.datePublished) : new Date();
+            break;
+        }
+    }
```

Comment:
You can use a selector like `script:contains("BlogPosting")` to select the required JSON-LD.